x/erc20/keeper/evm_hooks.go

```diff
@@ -63,104 +66,212 @@ func (k Keeper) PostTxProcessing(
+		k.Logger(ctx).Debug(
+			"ERC20 token -> Cosmos coin conversion is disabled for pair",
+			"coin", pair.Denom, "contract", pair.Erc20Address,
+		)
+		return
+	}
+
+	// create the corresponding sdk.Coin that is paired with ERC20
+	coins := sdk.Coins{{Denom: pair.Denom, Amount: math.NewIntFromBigInt(tokens)}}
+
+	// Perform token conversion. We can now assume that the sender of a
+	// registered token wants to mint a Cosmos coin.
+	switch pair.ContractOwner {
+	case types.OWNER_MODULE:
+		_, err = k.CallEVM(ctx, erc20, types.ModuleAddress, contractAddr, true, "burn", tokens)
+	case types.OWNER_EXTERNAL:
+		err = k.bankKeeper.MintCoins(ctx, types.ModuleName, coins)
+	default:
+		err = types.ErrUndefinedOwner
+	}
+
+	if err != nil {
+		k.Logger(ctx).Debug(
+			"failed to process EVM hook for ER20 -> coin conversion",
+			"coin", pair.Denom, "contract", pair.Erc20Address, "error", err.Error(),
+		)
+		return
+	}
+
+	// Only need last 20 bytes from log.topics
+	recipient := sdk.AccAddress(from.Bytes())
+
+	// transfer the tokens from ModuleAccount to sender address
+	if err := k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, recipient, coins); err != nil {
+		k.Logger(ctx).Debug(
+			"failed to process EVM hook for ER20 -> coin conversion",
+			"tx-hash", receipt.TxHash.Hex(), "log-idx", logid,
+			"coin", pair.Denom, "contract", pair.Erc20Address, "error", err.Error(),
+		)
+		return
+	}
+
+	// If a sender is converting tokens to coins, we want to track their address too
+	//
+	// NOTE: this only applies to native SDK coins and Wevmos
+	if (pair.IsNativeCoin() || isWevmos) && !k.HasSTRv2Address(ctx, from.Bytes()) {
+		k.SetSTRv2Address(ctx, from.Bytes())
+	}
+}
+
+func (k Keeper) PostProcessWithdrawDepositEvent(
+	ctx sdk.Context,
+	_ *ethtypes.Receipt,
+	log *ethtypes.Log,
+) {
+	wevmos := contracts.WEVMOSContract.ABI
+
+	// Check if event is included in WevmosContract
+	eventID := log.Topics[0]
+	event, err := wevmos.EventByID(eventID)
+	if err != nil {
+		return
+	}
+
+	// Check if event is a `Withdrawal` or Deposit event.
+	if event.Name != types.WevmosEventWithdraw && event.Name != types.WevmosEventDeposit {
+		return
+	}
+
+	withdrawEvent, err := wevmos.Unpack(event.Name, log.Data)
+	if err != nil {
+		return
+	}
+
+	if len(withdrawEvent) == 0 {
+		return
+	}
+
+	tokens, ok := withdrawEvent[0].(*big.Int)
+	// safety check and ignore if amount not positive
+	if !ok || tokens == nil || tokens.Sign() != 1 {
+		return
+	}
+
+	// Check that the contract is a registered token pair
+	contractAddr := log.Address
+
+	isWevmos := getIsWevmos(ctx, contractAddr)
+	if !isWevmos {
+		return
+	}
+
+	from := common.BytesToAddress(log.Topics[1].Bytes())
+
+	if !k.HasSTRv2Address(ctx, from.Bytes()) {
+		k.SetSTRv2Address(ctx, from.Bytes())
+	}
+}
+
+func getIsWevmos(ctx sdk.Context, contractAddr common.Address) bool {
+	switch {
+	case utils.IsMainnet(ctx.ChainID()):
+		return contractAddr.String() == erc20precompile.WEVMOSContractMainnet
```

Comment:
Nit: we should not hardcode contract addresses. This should be a module parameter instead

Reply:
I will make this change in the STR branch for v20!

Reply:
The wrapped coin address as a module param change is gonna be included in v20. 
For now its gonna be just the hardcoded addresses